### PR TITLE
refactor(sunjx): refactor dataset and reward module

### PR DESCRIPTION
**Pull request by Jiaxuan-Sun**
# 1. Dataset Module Refactoring (`lightrft/datasets/`)

**Modified:**
- `__init__.py`: Refactored imports with unified interfaces and improved optional dependency handling

**Added:**
- `config.py`: `DatasetConfig` class
  - Unified configuration for train/eval/pretrain datasets
  - Auto-normalization of `data_path` and `data_probs` (supports string/list)
  - Factory methods: `for_train()`, `for_eval()`, `for_pretrain()`
  - Parameter validation

- `loader.py`: `DatasetLoader` class
  - Unified loading interface for train/eval/pretrain datasets
  - Automatic handling of `blending_datasets` parameters
  - Support for `PromptDatasetVL` and `SFTDatasetVL`
  - Consistent logging

# 2. Reward Module (`lightrft/reward/`)

**Added:**
- `__init__.py`: Module entry point with unified exports

- `base.py`: `BaseReward` abstract base class
  - Unified `compute()` method signature
  - Consistent return format: `(rewards, metrics)`

- `rule.py`: `RuleReward` class
  - Rule-based reward implementation
  - Format checking (e.g., `<think>` tags, `\boxed{}` notation)
  - Accuracy verification using mathruler grader
  - Registry pattern for custom rule types
  - Built-in rules: `default`, `geo3k_*`, `gsm8k_*`

- `model.py`: Reward model implementations
  - `SingleRewardModel`: Single reward model wrapper with auto load/offload
  - `MultiRewardModel`: Multiple reward model ensemble with recipe-based aggregation
  - Supports standard PyTorch models and custom engines (e.g., SGLang)

- `manager.py`: `RewardManager` class
  - Unified manager for all reward types
  - Auto-selection of reward implementation (rule/single/multi)
  - `from_config()` factory method

### CODE REVIEW

File: lightrft/datasets/__init__.py [hunk 1]
```diff
@@ -1,21 +1,54 @@
...
+Main Features:
+    - Unified dataset configuration via DatasetConfig
+    - Consistent loading interface via DatasetLoader
+    - Support for train, eval, and pretrain datasets
+    - Automatic handling of blending_datasets parameters
+
+Classes:
+    DatasetConfig: Configuration class for dataset loading
+    DatasetLoader: Unified loader for all dataset types
+"""
```
Comment (puyuan1996) [inline lightrft/datasets/__init__.py]: We need to update the examples in gsm8k and geo3k to use the refactored dataset and reward APIs introduced here.

File: lightrft/datasets/__init__.py [hunk 2]
```diff
@@ -1,21 +1,54 @@
...
+Dataset Module for LightRLHF
```
Comment (puyuan1996) [inline lightrft/datasets/__init__.py]: LightRFT

**Review by puyuan1996 — CHANGES_REQUESTED** [review]
